src/test/java/HeapTest.java

```diff
@@ -15,75 +15,16 @@ class HeapTest {
-        Heap<Integer> singletonB = new Heap<>((Comparator<Integer>) Comparator.naturalOrder());
-        singletonA.add(10);
-        singletonB.add(10);
-
-        Heap<Integer> manyA = new Heap<>((Comparator<Integer>) Comparator.naturalOrder());
-        Heap<Integer> manyB = new Heap<>((Comparator<Integer>) Comparator.naturalOrder());
-        manyA.add(10);
-        manyA.add(20);
-        manyA.add(10);
-        manyA.add(5);
-        manyA.add(15);
-        manyB.add(10);
-        manyB.add(20);
-        manyB.add(10);
-        manyB.add(5);
-        manyB.add(15);
-
-        Heap<Integer> unequalDifferentValues = new Heap<>((Comparator<Integer>) Comparator.naturalOrder());
-        unequalDifferentValues.add(100);
-        unequalDifferentValues.add(200);
-        unequalDifferentValues.add(100);
-        unequalDifferentValues.add(50);
-        unequalDifferentValues.add(150);
-
-        Heap<Integer> unequalDifferentOrder = new Heap<>((Comparator<Integer>) Comparator.naturalOrder());
-        unequalDifferentOrder.add(5);
-        unequalDifferentOrder.add(10);
-        unequalDifferentOrder.add(10);
-        unequalDifferentOrder.add(15);
-        unequalDifferentOrder.add(20);
-
-        Heap<Integer> unequalDifferentSizes = new Heap<>((Comparator<Integer>) Comparator.naturalOrder());
-        unequalDifferentSizes.add(10);
-        unequalDifferentSizes.add(20);
-        unequalDifferentSizes.add(10);
-        unequalDifferentSizes.add(5);
-
-        Heap<Integer> unequalSomeEqual = new Heap<>((Comparator<Integer>) Comparator.naturalOrder());
-        unequalSomeEqual.add(10);
-        unequalSomeEqual.add(20);
-        unequalSomeEqual.add(10);
-        unequalSomeEqual.add(5);
-        unequalSomeEqual.add(20);
-
-        Heap<Integer> unequalDifferentComparator = new Heap<>((Comparator<Integer>) Comparator.naturalOrder()
-                .reversed());
-        unequalDifferentComparator.add(10);
-        unequalDifferentComparator.add(20);
-        unequalDifferentComparator.add(10);
-        unequalDifferentComparator.add(5);
-        unequalDifferentComparator.add(15);
-
-        new EqualsTester().addEqualityGroup(Heap.class)
-                .addEqualityGroup(emptyA, emptyB)
-                .addEqualityGroup(singletonA, singletonB)
-                .addEqualityGroup(manyA, manyB)
-                .addEqualityGroup(unequalDifferentValues)
-                .addEqualityGroup(unequalDifferentOrder)
-                .addEqualityGroup(unequalDifferentSizes)
-                .addEqualityGroup(unequalSomeEqual)
-                .addEqualityGroup(unequalDifferentComparator)
-                .testEquals();
+    private <T> boolean check_equals(Heap<T> heap1, Heap<T> heap2) {
+        if (heap1.size() != heap2.size()) {
+            return false;
+        }
+        while (!heap1.isEmpty()) {
+            if (!Objects.equals(heap1.remove(), heap2.remove())) {
+                return false;
+            }
+        }
```

Comment:
It would be a little better to copy the heap contents into a list, then assert equality on the list. Right now, when these tests fail, you'll only get failure messages like "expected false was true," which is completely useless. Asserting list equality would let you know that `[2,1] was not [1,2]`.

Reply:
I made the change and removed the equal check method and added a method to convert the heap to the list for comparison.